modal/functions.py

```diff
@@ -802,6 +802,16 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
             if cpu is not None and cpu < 0.25:
                 raise InvalidError(f"Invalid fractional CPU value {cpu}. Cannot have less than 0.25 CPU resources.")
             milli_cpu = int(1000 * cpu) if cpu is not None else 0
+            if memory and isinstance(memory, int):
+                memory_mb = memory
+                memory_mb_max = 0  # no limit
+            elif memory and isinstance(memory, tuple):
+                memory_mb, memory_mb_max = memory
+                if memory_mb_max < memory_mb:
+                    raise ValueError(f"Cannot specify a memory limit lower than request: {memory_mb_max} < {memory_mb}")
+            else:
+                memory_mb = 0
+                memory_mb_max = 0
```

Comment:
Kind of annoying that we need to have this logic duplicated :/

Reply:
Yeh I could have factored it out into a helper 😅 . Rule of threes.